### PR TITLE
Expose metrics of root coordinator

### LINKED ISSUE #7162: Expose metrics of RootCoord
see: [mep8-add-metrics-for-proxy](https://wiki.lfaidata.foundation/display/MIL/MEP+8+--+Add+metrics+for+proxy)

/assign

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>

issue: #7162

### CODE REVIEW

**Comment by scsven** [comment]
@congqixia Please review this PR.

File: internal/rootcoord/root_coord.go [hunk 1]
```diff
@@ -1926,3 +1933,69 @@ func (c *Core) SegmentFlushCompleted(ctx context.Context, in *datapb.SegmentFlus
...
+			zap.Int64("node_id", c.session.ServerID),
+			zap.String("req", req.Request),
+			zap.String("metric_type", metricType),
+			zap.Any("systemInfoMetrics", systemInfoMetrics), // TODO(dragondriver): necessary? may be very large
+			zap.Error(err))
+
+		return systemInfoMetrics, err
+	}
+
+	log.Debug("RootCoord.GetMetrics failed, request metric type is not implemented yet",
```
Comment (congqixia) [inline internal/rootcoord/root_coord.go:1989]: log message outofdate

**Review by congqixia — CHANGES_REQUESTED** [review]
Please fix the issues other the design ones. You could have other pr to fix design ones

**Comment by DragonDriver** [comment]
> Please fix the issues other the design ones. You could have other pr to fix design ones

I will change the response from string to byte slice in next pr.

**Comment by DragonDriver** [comment]
a little curious if there are any benchmarks that compares the efficiency of passing string or byte slice?

**Comment by congqixia** [comment]
> a little curious if there are any benchmarks that compares the efficiency of passing string or byte slice?

The main issue is:  Marshaling info result is byte slice; Converting byte slice into string needs to copy whole slice which consuming extra memory and cpu time.

**Comment by DragonDriver** [comment]
> > a little curious if there are any benchmarks that compares the efficiency of passing string or byte slice?
> 
> The main issue is: Marshaling info result is byte slice; Converting byte slice into string needs to copy whole slice which consuming extra memory and cpu time.

make sense. thx for pointing out this.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**